src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -135,7 +116,10 @@ static ov::threading::IStreamsExecutor::Config make_task_executor_config(const E
         default: OPENVINO_ASSERT(false, "[GPU] Can't create task executor: invalid host task priority value: ", priority);
     }
 
-    adjust_num_cores(task_executor_config);
+    task_executor_config.update_executor_config(task_executor_config._streams,
+                                                1,
+                                                task_executor_config._threadPreferredCoreType,
+                                                false);
```

Comment:
why do we need `update` method? can all required valid values be set via class ctor?